src/audio-worklet/migration/spn-recorder/app.js

```diff
@@ -0,0 +1,289 @@
+/**
+ * @license Copyright (c) Google. MIT License.
+ * @fileOverview Provides functionality for
+ * the ScriptProcessorNode Recording Demo.
+ */
+'use strict';
+
+import createLinkFromAudioBuffer from './exporter.mjs';
+
+const context = new AudioContext();
+
+// Arbitrary buffer size, not specific for a reason
+const BUFFER_SIZE = 256;
+let isRecording = false;
+let isMonitoring = false;
+let visualizationEnabled = true;
+let currentSample = [];
+let recordingLength = 0;
+
+// Wait for user interaction to initialize audio, as per specification
+document.addEventListener('click', (element) => {
+  init();
+  document.querySelector('#click-to-start').remove();
+}, {once: true});
+
+/**
+ * Defines overall audio chain and initializes all functionality
+ */
+async function init() {
+  if (context.state === 'suspended') {
+    await context.resume();
+  }
+
+  // Get user's microphone and connect it to the AudioContext
+  const micStream = await navigator.mediaDevices.getUserMedia({
+    audio: {
+      echoCancellation: false,
+      autoGainControl: false,
+      noiseSuppression: false,
+      latency: 0,
+    },
+  });
+
+  const micSourceNode = context.createMediaStreamSource(micStream);
+
+  // Prepare buffer for recording
+  const recordBuffer = context.createBuffer(
+      2,
+      // 10 seconds seems reasonable for demo purposes
+      context.sampleRate * 10,
+      context.sampleRate,
+  );
+  const spNode = setupScriptProcessor(recordBuffer);
+
+  const monitorNode = context.createGain();
+  const inputGain = context.createGain();
+  const medianEnd = context.createGain();
+
+  setupMonitor(monitorNode);
+  setupRecording(recordBuffer);
+  setupVisualizers();
+
+  micSourceNode
+      .connect(inputGain)
+      .connect(medianEnd)
+      .connect(spNode)
+      .connect(monitorNode)
+      .connect(context.destination);
+}
+
+/**
+ * Creates ScriptProcessor to record and track microphone audio.
+ * @param {AudioBuffer} recordBuffer
+ * @return {ScriptProcessorNode} ScriptProcessorNode to pass audio into.
+ */
+function setupScriptProcessor(recordBuffer) {
+  const processor = context.createScriptProcessor(BUFFER_SIZE);
+  currentSample = new Array(recordBuffer.numberOfChannels).fill([]);
```

Comment:
Didn't notice that this is a global - this example is supposed to be "bad" so it shouldn't matter much, but how can we change this so it doesn't live on a global scope?

Reply:
Replaced the global with a passthrough function